pic-sure-api-data/src/main/java/edu/harvard/dbmi/avillach/data/entity/NamedDataset.java

```diff
@@ -0,0 +1,82 @@
+package edu.harvard.dbmi.avillach.data.entity;
+
+import javax.json.Json;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(description = "A NamedDataset object containing query, name, user, and archived status.")
+@Entity(name = "named_dataset")
+@Table(uniqueConstraints = { 
+    @UniqueConstraint(name = "unique_queryId_user", columnNames = { "queryId", "user" })
+})
+public class NamedDataset extends BaseEntity {
+    @Schema(description = "The associated Query")
+    @OneToOne
+    @JoinColumn(name = "queryId")
+    private Query query;
```

Comment:
How does this work if the query gets deleted?

Reply:
The query table and the named_dataset table must both persist. If you try to delete a query from the query table that is linked to an entry in the named dataset table, sql returns this error. So it's not possible to have the query deleted and the linked named_dataset exist because of those table constraints.
```
ERROR 1451 (23000): Cannot delete or update a parent row: a foreign key constraint fails (`picsure`.`named_dataset`, CONSTRAINT `foreign_queryId` FOREIGN KEY (`queryId`) REFERENCES `query` (`uuid`))
```